code/game/objects/items/devices/chameleonproj.dm

```diff
@@ -49,9 +49,17 @@
 		return
 	if(istype(target, /obj/structure/falsewall))
 		return
+	if(target.alpha == 0)
+		return
+	if(!target.invisibility == 0)
+		return
 	if(iseffect(target))
 		if(!(istype(target, /obj/effect/decal))) //be a footprint
 			return
+	if(istype(target, /obj/structure/closet/cardboard/agent) || istype(target, /obj/structure/holosign))
```

Comment:
Pretty sure these use alpha ?

Reply:
Holosigns do, but the box has alpha = 255 when disturbed

Reply:
But then it will just look like a box ? Unless i'm missing something.

Reply:
For some reason I assumed that it would copy the box's procs, turns out that's not the case